src/interactive-window/editor-integration/codeGenerator.ts

```diff
@@ -168,22 +173,26 @@ export class CodeGenerator implements IInteractiveWindowCodeGenerator {
         }
     }
 
-    private extractStrippedLines(metadata: Pick<InteractiveCellMetadata, 'interactive' | 'id'>): {
+    private extractStrippedLines(
+        metadata: Pick<InteractiveCellMetadata, 'interactive' | 'id' | 'interactiveWindowCellMarker'>
+    ): {
         stripped: string[];
         trueStartLine: number;
-        firstExecutableLineIndex: number;
     } {
         const lines = splitMultilineString(metadata.interactive.originalSource);
-        // Compute the code that will really be sent to jupyter
-        const stripped = this.extractExecutableLines(metadata.interactive.originalSource);
+        // Compute the code that will really be sent to jupyter (including the cell marker)
+        const { lines: stripped } = this.extractExecutableLines(metadata.interactive.originalSource);
```

Comment:
Note: I've simplified the code by sending the cell marker as well to the Jupyter kernel.
This way we don't have to strip white space and don't have to take into account any leading white space having been stripped off by Jupyter (as we'll have a cell marker - except in the case where user uses `shift enter` when there's no cell marker - and tha'ts handled as well and we have tests for that too)

Reply:
Sounds good, I was about to ask about the shift-enter case. Sounds like it's covered.